src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseFeedbackCreationService.java

```diff
@@ -45,8 +45,24 @@ public class ProgrammingExerciseFeedbackCreationService {
 
     private static final Predicate<String> IS_PYTHON_EXCEPTION_LINE = line -> line.startsWith(PYTHON_EXCEPTION_LINE_PREFIX);
 
-    private static final List<String> TIMEOUT_EXCEPTIONS = Arrays.asList("org.junit.runners.model.TestTimedOutException", "java.util.concurrent.TimeoutException",
-            "org.awaitility.core.ConditionTimeoutException", "Timed?OutException");
+    private static final String TIMEOUT_EXCEPTIONS_REGEX = String.join("|", "org.junit.runners.model.TestTimedOutException", "java.util.concurrent.TimeoutException",
+            "org.awaitility.core.ConditionTimeoutException", "TimedOutException");
+
+    private static final String EXCEPTION_PREFIX_MESSAGE_REGEX = "[a-zA-Z0-9._ ]{1,5000}";
```

Comment:
`a-zA-Z_0-9` is also contained in the word character class. Would that

```
[\w. ]{1,5000}
```

work?

https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html

Reply:
Great idea! Thanks